### ISSUE TITLE
`dbgd_mem_write` ignores size

### ISSUE THREAD

**Issue by JayFoxRox** [mborgerson/nxdk-rdt#2]
For MMIO we should have control over the access size.
I assume one can put multiple requests in one packet, so writing of arrays / buffers is not necessary as we can just send a ton of small requests.

**Comment by mborgerson** [comment]
Yeah, there's a size field for exactly this reason. Just needs to be implemented.

**Comment by JayFoxRox** [comment]
Done in this branch: https://github.com/JayFoxRox/nxdk-rdt/pull/4

* It will try to read/write as many `uint32_t`, then `uint16_t` and then `uint8_t`
* Uses a bytes buffer for sending and receiving
